[Cocoa] Delete dead code after cleaning up HAVE() and USE() macros

litherum opened the pull request:
#### 6a90e53bcf696a35822fec8282cdc6b230732fef
<pre>
[Cocoa] Delete dead code after cleaning up HAVE() and USE() macros
<a href="https://bugs.webkit.org/show_bug.cgi?id=258436">https://bugs.webkit.org/show_bug.cgi?id=258436</a>
rdar://111198556

Reviewed by Tim Horton and Alexey Proskuryakov.

Now that we&apos;ve removed a bunch of configurations, there is a bunch of
leftover dead code we can remove.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/runtime/SymbolStubsForSafariCompatibility.mm: Removed.
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WTF/wtf/text/cocoa/StringCocoa.mm:
(WTF::String::String): Deleted.
* Source/WebCore/PAL/pal/cf/CoreTextSoftLink.cpp:
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:
* Source/WebCore/PAL/pal/spi/mac/HIToolboxSPI.h:
* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebCore/PAL/pal/spi/mac/QuickLookMacSPI.h:
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::createTextStyleFont):
(WebCore::mapWidth):
* Source/WebCore/platform/network/cf/NetworkStorageSessionCFNet.cpp:
(WebCore::NetworkStorageSession::createCFStorageSessionForIdentifier):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::createPrivateStorageSession):
* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::initializeSandbox):
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::NetworkDataTaskCocoa::NetworkDataTaskCocoa):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::NetworkRTCUDPSocketCocoaConnections::createNWConnection):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUtilitiesCocoa.mm:
(WebKit::setNWParametersApplicationIdentifiers):
(WebKit::setNWParametersTrackerOptions):
(WebKit::isKnownTracker):
* Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
* Source/WebKit/Platform/spi/Cocoa/NWSPI.h:
* Source/WebKit/Platform/spi/mac/ApplicationServicesSPI.h:
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::initializeSandbox):
* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:

Canonical link: <a href="https://commits.webkit.org/265448@main">https://commits.webkit.org/265448@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aba13a2dcfb8620e7c436e3065eddaff83dee182

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10924 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11137 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11451 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12571 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10940 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13514 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11118 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/13362 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11083 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11983 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9203 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12975 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9276 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9866 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17096 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9244 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10347 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/10018 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13256 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10357 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10460 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8549 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11033 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9635 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3016 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2622 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13901 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11342 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10318 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2782 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
```diff
@@ -29,7 +29,7 @@
 #include <AccessibilitySupport.h>
 #endif
 
-#if !USE(APPLE_INTERNAL_SDK) || !HAVE(APP_SPECIFIC_ACCESSIBILITY_SETTINGS)
```
Did you remove it from PlatformHave.h in an earlier change? I don't see it being removed here.

litherum replied:
Good catch! I managed to delete all the uses of it but not the original definition.

litherum commented on Source/WTF/wtf/WTFConfig.cpp:
```diff
@@ -76,23 +76,13 @@ void setPermissionsOfConfigPage()
         mach_vm_address_t addr = bitwise_cast<uintptr_t>(static_cast<void*>(WebConfig::g_config));
         auto flags = VM_FLAGS_FIXED | VM_FLAGS_OVERWRITE;
-#if HAVE(VM_FLAGS_PERMANENT)
-        flags |= VM_FLAGS_PERMANENT;
```
This is wrong

litherum commented on Source/WTF/wtf/WTFConfig.cpp:
```diff
@@ -76,23 +76,13 @@ void setPermissionsOfConfigPage()
         auto result = attemptVMMapping();
 
-#if HAVE(VM_FLAGS_PERMANENT)
-        if (result != KERN_SUCCESS) {
```
Ditto

litherum committed:
[Cocoa] Delete dead code after cleaning up HAVE() and USE() macros
https://bugs.webkit.org/show_bug.cgi?id=258436
rdar://111198556

Reviewed by Tim Horton and Alexey Proskuryakov.

Now that we've removed a bunch of configurations, there is a bunch of
leftover dead code we can remove.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/runtime/SymbolStubsForSafariCompatibility.mm: Removed.
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WTF/wtf/text/cocoa/StringCocoa.mm:
(WTF::String::String): Deleted.
* Source/WebCore/PAL/pal/cf/CoreTextSoftLink.cpp:
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:
* Source/WebCore/PAL/pal/spi/mac/HIToolboxSPI.h:
* Source/WebCore/PAL/pal/spi/mac/NSWindowSPI.h:
* Source/WebCore/PAL/pal/spi/mac/QuickLookMacSPI.h:
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::createTextStyleFont):
(WebCore::mapWidth):
* Source/WebCore/platform/network/cf/NetworkStorageSessionCFNet.cpp:
(WebCore::NetworkStorageSession::createCFStorageSessionForIdentifier):
* Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm:
(WebCore::createPrivateStorageSession):
* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::initializeSandbox):
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::NetworkDataTaskCocoa::NetworkDataTaskCocoa):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::NetworkRTCUDPSocketCocoaConnections::createNWConnection):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUtilitiesCocoa.mm:
(WebKit::setNWParametersApplicationIdentifiers):
(WebKit::setNWParametersTrackerOptions):
(WebKit::isKnownTracker):
* Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
* Source/WebKit/Platform/spi/Cocoa/NWSPI.h:
* Source/WebKit/Platform/spi/mac/ApplicationServicesSPI.h:
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::initializeSandbox):
* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:

Canonical link: https://commits.webkit.org/265448@main
